lambda_functions/analyzer/binary_info.py

```diff
@@ -71,13 +72,9 @@ def __enter__(self):
         return self
 
     def __exit__(self, exception_type, exception_value, traceback):
-        """Remove the downloaded binary from local disk."""
-        # In Lambda, "os.remove" does not actually remove the file as expected.
-        # Thus, we first truncate the file to set its size to 0 before removing it.
-        if os.path.isfile(self.download_path):
-            with open(self.download_path, 'wb') as file:
-                file.truncate()
-            os.remove(self.download_path)
+        """Shred the downloaded binary and delete it from disk."""
+        # Note: This runs even during exception handling (it is the "with" context).
+        subprocess.check_call(['shred', '-u', self.download_path])
```

Comment:
only suggestion would be using the more verbose `--remove` flag instead of the shorthand `-u` so the purpose of this command is more explicit, but totally not necessary.

Reply:
Great idea! I'm always a huge fan of better readability.
Done!